src/rest.jl

```diff
@@ -81,7 +81,64 @@ function get_access_token(ctx::Context, creds::ClientCredentials)::AccessToken
     opts = (redirect = false, retry_non_idempotent = true, connect_timeout = 30, readtimeout = 30, keepalive = true)
     rsp = HTTP.request("POST", url, h, body; opts...)
     data = JSON3.read(rsp.body)
-    return AccessToken(data.access_token, data.scope, data.expires_in, now())
+    return AccessToken(data.access_token, data.scope, data.expires_in, datetime2unix(now()))
+end
+
+# cache name
+function _cache_file()
+    return joinpath(homedir(), ".rai", "tokens.json")
+end
+
+# read oauth cache
+function _read_cache()
+    try
+        if isfile(_cache_file())
+            return copy(JSON3.read(read(_cache_file())))
+        else
+            return nothing
+        end
+    catch e
+        @warn e
+        return nothing
+    end
+end
+
+# Read access token from cache
+function _read_token_cache(creds::ClientCredentials)
+    try
+        cache = _read_cache()
+        cache === nothing && return nothing
+
+        if creds.client_id in keys(cache)
```

Comment:
```suggestion
        if haskey(cache, creds.client_id)
```
should we also add an expiration check here and return `nothing` if the cached token is expired? I can imagine the workflow where you infrequently run things, this token may end up being stale, so it seems better to force the request of a new one.

But maybe we already do that check later, so it doesn't matter here?

Reply:
you are right, the workflow is the following: we get the cache -> check if token expired (fetch new one and save it to cache if expired) -> proceed